src/mam/mod_mam_rdbms_arch_async.erl

```diff
@@ -95,26 +95,36 @@ flush(Acc, Extra = #{host_type := HostType, queue_length := MessageCount}) ->
 do_flush_pm(Acc, #{host_type := HostType, queue_length := MessageCount,
                    batch_size := MaxSize, batch_name := BatchName}) ->
     Rows = [mod_mam_rdbms_arch:prepare_message(HostType, Params) || Params <- Acc],
-    InsertResult =
-        case MessageCount of
-            MaxSize ->
-                mongoose_rdbms:execute(HostType, BatchName, lists:append(Rows));
-            OtherSize ->
-                Results = [mongoose_rdbms:execute(HostType, insert_mam_message, Row) || Row <- Rows],
-                case lists:keyfind(error, 1, Results) of
-                    false -> {updated, OtherSize};
-                    Error -> Error
-                end
-        end,
-    case InsertResult of
-        {updated, _Count} -> ok;
-        {error, Reason} ->
-            mongoose_metrics:update(HostType, modMamDropped, MessageCount),
-            ?LOG_ERROR(#{what => archive_message_failed,
-                         text => <<"archive_message query failed">>,
-                         message_count => MessageCount, reason => Reason}),
-            ok
+    IsFullBuffer = MessageCount =:= MaxSize,
```

Comment:
This code is exactly the same as for the other module and I suppose there are more places where `mod_mam` and `mod_mam_muc` have duplicated code. I'm aware that such refactor is out of scope of this PR. Maybe we should create an issue in backlog for unifying common code in them?

Reply:
The key is the name of the prepared query that will be executed for example, so the code is very parallel but there's plenty of names around that aren't the same 😕